server/executor/runner.go

```diff
@@ -124,13 +127,19 @@ func (r persistentRunner) processExecQueue(job execReq) {
 		panic(err)
 	}
 
-	response, err := triggerer.Trigger(job.ctx, job.test, job.run.TraceID, job.run.SpanID)
+	triggerSpanCtx, span := r.triggerSpanTracer.Start(context.Background(), "Tracetest trigger")
```

Comment:
If you move this to the `instrumentedTriggerer.Trigger` func, you wouldn't need to use 2 contexts in the `Trigger` interface. Given that no trigger type is actually using both contexts, I understand that this is passed just because the`instrumentedTriggerer` doesn't have the `triggerSpanTracer` dependency. I'd rather add a dependency to the `instrumentedTriggerer` than to have 2 contexts on an interface that aren't actually required

Reply:
Not only that, to do that we either have to:

* Add a pointer to the test run to the Trigger interface
* Change the trigger response object to have both traceID and spanID

Without either of those, we can't update the run object with the correct traceID and spanID.